merino-cache/src/memory.rs

```diff
@@ -106,13 +108,18 @@ impl Suggester {
         provider: Box<dyn SuggestionProvider>,
         metrics_client: StatsdClient,
     ) -> Box<Self> {
-        let items = Arc::new(DedupedMap::new());
+        let this = Self {
```

Comment:
I'm very open to better names that `this` and `that` for the two copies of the suggester object. Note `self` is a reserved word, and cannot be used here.

Reply:
Maybe use `suggester` and `cloned`?